meilisearch-release-changelog/v1.8.0.md

```diff
@@ -0,0 +1,219 @@
+}
+
+```
+
+### Expose the `distribution` shift setting
+
+When setting an embedder, you can now set the `distribution` shift.
+
+```json5
+"default": {
+  "source": "huggingFace", // supported for any source
+  "model": "some/model",
+  "distribution": {  // describes the natural distribution of results
+    "mean": 0.7, // mean value
+    "sigma": 0.3 // variance
+  }
+}
+```
+
+The “distribution shift” is an affine transformation applied to the `_rankingScore` of a **semantic** search result with the aim of making the comparison to the `_rankingScore` of a **keyword** search result more meaningful.
+
+### Other hybrid search improvements
+
+* Hide the API key in settings and task queue (#4533) @dureuill
+* Return the keyword search results even in case of a failure of the embedding (#4548) @dureuill
+* For hybrid or semantic search requests, add a `semanticHitCount` field at the top of the search response indicating the number of hits originating from the semantic search (#4548) @dureuill
+
+## Support negative keyword when searching
+
+Search queries can now contain a negative keyword to exclude terms from the search. Use the `-` operator in front of a word or a phrase to make sure no document that contains those words are shown in the results.
+
+- `-escape` returns a placeholder search without any document contains the `escape` word.
+- `-escape room` returns only documents containing the `room` word but not the `escape` one.
+- `-"on demand"` returns a placeholder search but without any document containing the `"on demand"` phrase.
+
+Done by @Kerollmops in #4535.
+
+## Search robustness improvements
+
+### Add a search cutoff
+
+To avoid any crash and performance issues, Meilisearch now stops search requests lasting more than 150ms.
+
+If you want to customize this value, you can update the `searchCutoffMs` settings (value in ms):
+
+```bash
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "searchCutoffMs": 50
+  }'
+```
+
+The default value of the `searchCutoffMs` setting is `null` and corresponds to 150ms.
+
+Done by @irevoire in #4466.
+
+### Limit concurrent search requests
+
+Meilisearch now limits the number of search requests waiting to be processed to avoid consuming an unbounded amount of RAM and crashing. So a queue of search requests waiting to be processed has been introduced.
+
+👉 This change does NOT impact the search performance, but only the number of enqueued search requests to prevent from any security issues.
+
+The default number of requests in the queue is 1000.
+
+To change this limit, use the experimental CLI flag:
+
+```bash
+./meilisearch --experimental-search-queue-size 100
+```
+
+🗣️ This is an experimental flag and we need your help to improve it! Share your thoughts and feedback on this [GitHub discussion](https://github.com/orgs/meilisearch/discussions/729).
```

Comment:
```suggestion
### Search request queue

This release introduces a queue for search requests to prevent Meilisearch from consuming an unbounded amount of RAM and crashing.

The default number of requests in the queue is 1000. Relaunch your self-hosted instance with `--experimental-search-queue-size` to change this limit:

```bash
./meilisearch --experimental-search-queue-size 100
```

👉 This limit does NOT impact the search performance. It only affects the number of enqueued search requests to prevent security issues.

🗣️ This is an experimental feature and we need your help to improve it! Share your thoughts and feedback on this [GitHub discussion](https://github.com/orgs/meilisearch/discussions/729).
```